addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/MongoDbInjection.java

```diff
@@ -0,0 +1,687 @@
+        }
+        // injection attack to $Where and $mapReduce cluases
+        // The $where clause executes associated js function one time for each tuple --> sleep time
+        // = interval * nTuples
+        if (doTimedScan) {
+            if (isStop()) {
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug(STOP_LOG);
+                }
+                return;
+            }
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("\nStarting with the javascript code injection package of attack rules:");
+            }
+            int aveRtt = getAveRtts(rtt),
+                    index = 0,
+                    timeShort = SLEEP_SHORT_TIME,
+                    timeLong = SLEEP_LONG_TIME;
+            String phase = null;
+            boolean hadTimeout = false;
+            String sleepValueToInj;
+            while (index < SLEEP_INJECTION.length) {
+                if (isStop()) {
+                    if (LOG.isDebugEnabled()) {
+                        LOG.debug(STOP_LOG);
+                    }
+                    return;
+                }
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("\nTrying  with the value: " + SLEEP_INJECTION[index][0]);
+                }
+                try {
+                    msgInjAttack = getNewMsg();
+                    sleepValueToInj =
+                            SLEEP_INJECTION[index][0].replaceFirst(
+                                    INSERT_SHORT_TIME, Integer.toString(timeShort));
+                    phase = INSERT_SHORT_TIME;
+                    setParameter(msgInjAttack, param, sleepValueToInj);
+                    sendAndReceive(msgInjAttack, false);
+                    if (LOG.isDebugEnabled()) {
+                        LOG.debug(
+                                "\nTrying for a longer time with the value: "
+                                        + SLEEP_INJECTION[index][1]);
+                    }
+                    if (msgInjAttack.getTimeElapsedMillis() >= aveRtt + SHORT_TRESHOLD) {
+                        phase = INSERT_LONG_TIME;
+                        sleepValueToInj =
+                                SLEEP_INJECTION[index][1].replaceFirst(
+                                        INSERT_LONG_TIME, Integer.toString(timeLong));
+                        msgInjAttack = getNewMsg();
+                        setParameter(msgInjAttack, param, sleepValueToInj);
+                        sendAndReceive(msgInjAttack, false);
+                        if (msgInjAttack.getTimeElapsedMillis() >= aveRtt + LONG_TRESHOLD) {
+                            bingo(
+                                    Alert.RISK_HIGH,
+                                    Alert.CONFIDENCE_HIGH,
+                                    getName(),
+                                    getDescription(),
+                                    null,
+                                    param,
+                                    SLEEP_INJECTION[index][1],
```

Comment:
This should use `sleepValueToInj` to have the time replaced? (Same for following bingo.)